backend/utils/ssl/client.go

```diff
@@ -201,6 +203,13 @@ func (c *AcmeClient) UseDns(dnsType DnsType, params string, websiteSSL model.Web
 		huaweiCloudConfig.PollingInterval = pollingInterval
 		huaweiCloudConfig.TTL = int32(ttl)
 		p, err = huaweicloud.NewDNSProviderConfig(huaweiCloudConfig)
+	case RainYun:
+		rainyunConfig := rainyun.NewDefaultConfig()
+		rainyunConfig.APIKey = param.APIkey
+		rainyunConfig.PropagationTimeout = propagationTimeout
+		rainyunConfig.PollingInterval = pollingInterval
+		rainyunConfig.TTL = ttl
+		p, err = rainyun.NewDNSProviderConfig(rainyunConfig)
 	}
 	if err != nil {
 		return err
```

Comment:
The provided code snippet appears to be adding support for the Rain Yun DNS provider in a Go application using the `go-acme/lego` library. It includes several changes:

1. **Import Statement**: A new import statement has been added for the Rain Yun DNS provider package (`rainyun`) at line 3.

2. **DNS Param Structure Update**: The `param` variable is now expected to include fields specific to the Rain Yun DNS provider, such as API key, propagation timeout, polling interval, and TTL.

3. **DNS Provider Configurations**:
   - For the Hong Kong server of Godaddy: Configuration settings are added for `hostname`, `propagationTimeout`, `pollingInterval`, and `ttl`.
   - For the mainland China server of Tencent Cloud: Similarly, configuration settings are added.
   - For Huawei Cloud: Similar configurations are applied.
   - For Rain Yun: Additional parameters like `APIkey`, `propagationTimeout`, `pollingInterval`, and `TTL` are required when registering with this service.

These updates should ensure that the application can handle requests specifically targeting Rain Yun's DNS infrastructure while maintaining compatibility with other existing DNS services managed by similar providers. If you encounter any issues during implementation or have further questions, feel free to ask!